pypeman/plugins/remoteadmin/views.py

```diff
@@ -0,0 +1,241 @@
+
+    :params channel: The channel name to stop.
+    """
+    if not ws:
+        ws = web.WebSocketResponse()
+        await ws.prepare(request)
+
+    chan = get_channel(channelname)
+    await chan.stop()
+
+    resp_message = json.dumps({
+        'name': chan.name,
+        'status': channels.BaseChannel.status_id_to_str(chan.status)
+    })
+    await ws.send_str(resp_message)
+
+
+async def list_msgs(request, channelname, ws=None):
+    """
+    List first `count` messages from message store of specified channel.
+
+    :params channel: The channel name.
+    """
+    if ws is None:
+        ws = web.WebSocketResponse()
+        await ws.prepare(request)
+
+    chan = get_channel(channelname)
+
+    args = request.rel_url.query
+    start = int(args.get("start", 0))
+    count = int(args.get("count", 10))
+    order_by = args.get("order_by", "timestamp")
+    start_dt = args.get("start_dt", None)
+    end_dt = args.get("end_dt", None)
+    text = args.get("text", None)
+    rtext = args.get("rtext", None)
+
+    messages = await chan.message_store.search(
+        start=start, count=count, order_by=order_by, start_dt=start_dt, end_dt=end_dt,
+        text=text, rtext=rtext) or []
+
+    for res in messages:
+        res['timestamp'] = res['message'].timestamp_str()
+        res['message'] = res['message'].to_json()
+
+    resp_message = json.dumps({'messages': messages, 'total': await chan.message_store.total()})
+    await ws.send_str(resp_message)
+
+
+async def replay_msg(request, channelname, message_id, ws=None):
+    """
+    Replay messages from message store.
+
+    :params channel: The channel name.
+    :params msg_ids: The message ids list to replay.
+    """
+    if not ws:
+        ws = web.WebSocketResponse()
+        await ws.prepare(request)
+
+    chan = get_channel(channelname)
+    result = []
+    try:
+        msg_res = await chan.replay(message_id)
+        result.append(msg_res.to_dict())
+    except Exception as exc:
+        result.append({'error': str(exc)})
+
+    resp_message = json.dumps(result)
+    await ws.send_str(resp_message)
+
+
+async def view_msg(request, channelname, message_id, ws=None):
+    """
+    Permit to get the content of a message
+
+    :params channel: The channel name.
+    :params msg_ids: The message ids list to replay.
+    """
+    if not ws:
+        ws = web.WebSocketResponse()
+        await ws.prepare(request)
+
+    chan = get_channel(channelname)
+    result = []
+    try:
+        msg_res = await chan.message_store.get_msg_content(message_id)
+        result.append(msg_res.to_dict())
+    except Exception as exc:
+        result.append({'error': str(exc)})
+
+    resp_message = json.dumps(result)
+    await ws.send_str(resp_message)
+
+
+async def preview_msg(request, channelname, message_id, ws=None):
+    """
+    Permits to get the 1000 chars of a message payload
+
+    :params channel: The channel name.
+    :params msg_ids: The message ids list to replay.
```

Comment:
to replay -> to preview ?

Reply:
i changed comment here and i rewrote many others